### ISSUE TITLE
Should we drop `test_dot_product_spill` (formerly `test_tensordot_stress`)?

### ISSUE THREAD

**Issue by fjetter** [coiled/coiled-runtime#662]
The benchmark `test_dot_product_spill` is implementing a tensordot (highly non-linear all-to-all network computation) with relatively large data. The test itself is designed such that the cluster inevitably requires spill-to-disk to complete the workload given the current algorithm.

For context, there are
- `test_dot_product` which is effectively performing the same operation but on a smaller dataset s.t. spilling is not triggered.
- `test_spilling` which specifically stresses the spill-to-disk implementation w/out network or compute (This is also slow and problematic IMO).

I'm inclined to drop this tensordot+spill test entirely but would like to get some input before making this decision.

The Pro/Con list is not necessarily exhaustive but these are the most relevant points on my end

### Pro
- This workload is stressing the entire system. It generates a relatively large number of tasks, uses spill-to-disk, a lot of network and a non-trivial amount of actual compute. Chances are that if a regression is introduced in any of our subsystems, this will be impacted
- It places the system under very heavy stress and is a good proxy to test whether dask operates stable if we reach hardware limits

### Con
- Tensordot is a relatively rare operation. We will very likely not optimize this in the foreseeable future. 
- The test runtimes are very large. They are currently taking about 320-400s and are parameterized to run twice totaling at about 12min. That's more than 20% of our total CI runtime.
- Tensordot plus spilling is apparently a workload that is not very common. Folks that are using tensordot typically try to avoid spilling (https://github.com/coiled/coiled-runtime/issues/652#issuecomment-1375792634)
- I don't really care how fast this is as long as it doesn't fail.

In conclusion the value of this test for me is not necessarily about the tensordot or the spilling but rather about hitting the system with a sledge hammer to see if it falls over. I believe this is a very valuable thing to test but I'm not necessarily convinced that paying 20% in our benchmarking suite for this is worth it. I'm also not interested at all in how fast this actually is but am rather interested in whether it is successful or not. This is a test that should be part of a "stress test" or "integration test" suite but not be part of benchmarking.

Thoughts?

cc @crusaderky @mrocklin

**Comment by crusaderky** [comment]
> Tensordot is a relatively rare operation. 
> Tensordot plus spilling is apparently a workload that is not very common.

I don't care about tensordot specifically. However, I believe it is highly representative of a generic O(n^2) problem:

1. memory usage per worker is directly influenced by threads per worker
2. memory usage explodes quadratically as you increase the dataset size, so that it quickly becomes unviable to mount enough physical RAM to fit the problem as you scale up
3. the same chunks of data are, at the same time, requested both by the next step of compute on the same worker and by many other workers on the cluster
4. it is highly representative for e.g. https://github.com/dask/distributed/issues/5996
5. Unlike `test_spilling`, it demonstrates a high CPU load from the tasks (good) while you're busy spilling/unspilling

>  I'm also not interested at all in how fast this actually is 

@ntabris do you have a feeling of how frequent heavy spilling activity is for real customers?

**Comment by crusaderky** [comment]
P.S. let's not forget *survivor bias*.
If we're not seeing many real life very large O(n^2) workloads on dask, it can mean two things:
- very large O(n^2) problems are not a common or interesting thing to solve for most people
- most people that need to solve them tried dask, failed hard or found it impossibly slow or expensive, and moved to a different tool
